packages/medusa/src/services/line-item.js

```diff
@@ -90,45 +90,50 @@ class LineItemService extends BaseService {
   }
 
   async generate(variantId, regionId, quantity, config = {}) {
-    const variant = await this.productVariantService_.retrieve(variantId, {
-      relations: ["product"],
-    })
-
-    const region = await this.regionService_.retrieve(regionId)
-
-    let price
-    let shouldMerge = true
-
-    if (config.unit_price && typeof config.unit_price !== `undefined`) {
-      // if custom unit_price, we ensure positive values
-      // and we choose to not merge the items
-      shouldMerge = false
-      if (config.unit_price < 0) {
-        price = 0
+    return this.atomicPhase_(async manager => {
+      const variant = await this.productVariantService_
+        .withTransaction(manager)
+        .retrieve(variantId, {
+          relations: ["product"],
+        })
+
+      const region = await this.regionService_
+        .withTransaction(manager)
+        .retrieve(regionId)
+
+      let price
+      let shouldMerge = true
+
+      if (config.unit_price && typeof config.unit_price !== `undefined`) {
+        // if custom unit_price, we ensure positive values
+        // and we choose to not merge the items
+        shouldMerge = false
+        if (config.unit_price < 0) {
+          price = 0
+        } else {
+          price = config.unit_price
+        }
```

Comment:
Same as previous comment. If meant to be in this PR, I think we should add a comment in the description or move it to an other.

Reply:
See comment above